media/js/teach/smarton.js

```diff
@@ -0,0 +1,303 @@
+
+            window.dataLayer.push({
+                event: 'smarton-interactions',
+                interaction: 'dismiss sticky footer',
+                footer: footerVisible
+            });
+        });
+    }
+
+    // Change the navbar current item to match the section waypoint
+    function navState(current, previous) {
+        return function(direction) {
+            if (fixed) {
+                if (direction === 'down') {
+                    $header.find('.nav-steps li').removeClass();
+                    $('#nav-step-' + current).addClass('current');
+                }
+                else {
+                    $header.find('.nav-steps li').removeClass();
+                    $('#nav-step-' + previous).addClass('current');
+                }
+            }
+        };
+    }
+
+    if ($header.length > 0) {
+        //Fire the waypoints for each section, passing classes for the current and previous sections
+        $('#ask').waypoint(navState('ask', 'ask'), { offset: $header.height() });
+        $('#know').waypoint(navState('know', 'ask'), { offset: $header.height() });
+        $('#do').waypoint(navState('do', 'know'), { offset: $header.height() });
+        $('#chat').waypoint(navState('chat', 'do'), { offset: $header.height() });
+    }
+
+    // Draw animated SVG pie charts
+    // Requires /libs/circles.min.js - http://lugolabs.com/circles
+    function drawCircle(circleId, circleVal){
+        Circles.create({
+          id:                  circleId,
+          value:               circleVal,
+          radius:              70,
+          maxValue:            100,
+          width:               12,
+          text:                function(value){return value + '%';},
+          duration:            850,
+          wrpClass:            'circles-wrap',
+          textClass:           'circles-text',
+          valueStrokeClass:    'circles-valueStroke',
+          maxValueStrokeClass: 'circles-maxValueStroke',
+          styleWrapper:        false,
+          styleText:           false,
+        });
+    }
+
+    // Trigger the chart when it scrolls into view
+    $('.sidebar-chart').each(function(){
+        var stage = $(this).children('.chart-stage');
+        var circleId = stage.attr('id');
+        var circleVal = stage.data('chart-value');
+        var waypoint = new Waypoint({
+            element: stage,
+            handler: function(direction) {
+                if (direction == 'down') {
+                    drawCircle(circleId, circleVal);
+                }
+            },
+            offset: '100%'
+        });
+    });
+
+    // Play videos in a modal and track the clicks
+    $('a.video-open').attr('role', 'button').on('click', function(e) {
+        e.preventDefault();
+
+        var $this = $(this);
+        var embedSrc = $this.data('embed-src');
+        var embedWrap = $('#' + $this.data('element-id'));
+
+        Mozilla.Modal.createModal(this, embedWrap, {
```

Comment:
Yeah, it's still not perfect but may be the best we can do. They don't want to send people away from our site and opening a new tab might be even worse. 

I'm writing in the iframe src when the modal opens to prevent the page from loading while it's hidden, hence why it starts over if you close and reopen the modal, but I do need to also destroy it when it closes so the loading can stop.